Sources/AuthFoundation/Network/APIClientError.swift

```diff
@@ -36,7 +36,7 @@ public enum APIClientError: Error {
     case unsupportedContentType(_ type: APIContentType)
     
     /// Received the given HTTP error from the server.
-    case serverError(_ error: Error)
+    case httpError(_ error: Error)
```

Comment:
why make this change? usually we get more granular errors than just HTTP codes, for example `EXXXXXXX` https://developer.okta.com/docs/reference/error-codes/

Reply:
There's confusion about the name `serverError` across different error types.  This is for the low-level `APIClient` interface, and represents the error we get back from URLSession.  Elsewhere, the OAuth2Error has a `serverError` type which returns an instance of [OAuth2ServerError](https://okta.github.io/okta-mobile-swift/development/authfoundation/documentation/authfoundation/oauth2servererror) that exposes the code, description, etc returned from the server.